src/Adapter/Driver/Mysqli/Result.php

```diff
@@ -51,9 +51,9 @@ class Result implements
     protected $nextComplete = false;
 
     /**
-     * @var bool
+     * @var mixed
```

Comment:
Do we have any stricter type that can be used here?

Reply:
This can be `null` or `array`, depending on the `current()` state. Moreover, I just reviewed all the adapter results and this is the scenario:
```
Adapter\Driver\Pgsql - current(): array|false
Adapter\Driver\Mysqli - current(): array|null
Adapter\Driver\Oci8 - current(): array|false
Adapter\Driver\Pdo - current(): array|object|false
Adapter\Driver\Sqlsrv - current(): array|false|null
```
I think we should try to normalize it. Maybe, we can throw an Exception if the adapter returns an error during the fetch process, and use `null` as empty value. Of course, this will be released with zend-db 3.0.0
@Ocramius and @willy0275 what do you think?